nixos/lib/testing/nodes.nix

```diff
@@ -159,10 +160,17 @@ in
 
     passthru.nodes = config.nodesCompat;
 
-    defaults = mkIf config.node.pkgsReadOnly {
-      nixpkgs.pkgs = config.node.pkgs;
-      imports = [ ../../modules/misc/nixpkgs/read-only.nix ];
-    };
+    defaults = mkMerge [
+      (mkIf (config.node.pkgs != null) {
+        nixpkgs.pkgs = config.node.pkgs;
+      })
+      (mkIf (config.node.pkgs == null) {
+        nixpkgs.pkgs = mkDefault hostPkgs;
+      })
```

Comment:
I'm pretty sure this overrides the normal `nodes.<name>.nixpkgs.config` behavior, silently.

```suggestion
```

Reply:
Done